Visual enhancement of the navigation bar and addition of a search bar for projects

tgrl84 opened the pull request:
Added a search bar for projects. 
Enhanced the navigation bar by integrating the search bar for a cleaner look.
Added the healthchecks name on projects page.
Adjusted the width for smoother transitions between pages.
Before :
![before_projects](https://github.com/healthchecks/healthchecks/assets/71886156/c718b281-1ba3-44be-bab7-779f6888521f)
![before_checks](https://github.com/healthchecks/healthchecks/assets/71886156/a38d4156-d0b8-49f6-88d9-b1c1103f29ef)
After :
![after_projects](https://github.com/healthchecks/healthchecks/assets/71886156/a8128a72-2934-455f-baa0-f761611687d9)
![after_checks](https://github.com/healthchecks/healthchecks/assets/71886156/b9c731ea-f383-426c-83ad-3c6a8d97462d)

cuu508 commented:
Thanks for the PR, @tgrl84!

I think it is a good idea to have a search bar in the projects page. There are indeed users with lots of projects, where this would be useful.

I would prefer to keep the search bar in the current place, below main navigation:

* It logically belongs to the page, not to the global navigation. It is for filtering content displayed on the current page, not for site-wide search.
* And there's a space for it: in the checks page, we need space to display tags, so the search bar can share the space. In the projects page we have the "My Projects" heading, and the search bar can share the space.

Would you be able to move the search bar to the previous location? 
Thanks!

tgrl84 commented:
Thanks for your feedback, I've moved the search bar back to its previous position as per your suggestion. I've updated the pull request with these changes.

cuu508 commented on templates/base.html:
```diff
@@ -86,6 +86,17 @@
                 <span class="icon-bar"></span>
             </button>
 
+            <a id="base-url" class="navbar-brand" href="{% url 'hc-index' %}">
```
With this change, the logo is shown unconditionally, also in the list of checks page, project's settings page, check's details page etc.

I prefer to leave things as is: have branding on the "public" pages, and on the project overview, but not elsewhere. It adds visual clutter and serves no purpose.

cuu508 commented on templates/base.html:
```diff
@@ -103,19 +114,13 @@
-                    alt="{{ site_name }}">
-            </a>
             {% endif %}
+            {% if not project %}
```
Same as the previous content, I'd prefer to revert this.

tgrl84 commented on static/css/base.css:
```diff
@@ -148,13 +149,13 @@ body {
-    /* Fluid below 1320px, but max width capped to 1320px ... */
-    max-width: 1320px;
+    /* Fluid below 1170px, but max width capped to 1170px ... */
+    max-width: 1170px;
```
I understand that but then why don't put all the pages in this width instead of changing between each pages that make transitions look weird

cuu508 replied:
IIRC I started with Bootstrap's default width for all pages. I needed to make the pages wider in a few places to make the content fit better, but didn't want to make the change globally, because then I would need to review all other pages and fix instances where the  wider layout looks bad (for example, because there's not much content on the page). So I made selective changes in just a few pages.

tgrl84 commented on static/css/base.css:
```diff
@@ -6,6 +6,7 @@ html {
 body {
     /* Margin bottom by footer height */
     margin-bottom: 84px;
+    overflow-y: scroll;
```
Just for the transition between pages which looked weird before, it goes with the width change

cuu508 replied:
Sorry, but I still do not understand why precisely this change is needed.

cuu508 commented:
@tgrl84 are you planning to continue work on this, or should I take over?
